### PR TITLE
Migrate Bittrex integration to API v3

### PR DESCRIPTION
**Pull request by Exef**
TODOs: 
- [x] Gather real data for mocked tests 

Points to double-check:
1. In the [upgrade guide](https://bittrex.github.io/guides/v3/upgrade) it stated that `Market symbols are reversed in v3 to align with standard forex conventions`. I need to clarify if it's needed to change the order of the provided `pair`.
2. In the V3 `QuantityRemaining` is replaced with `fillQuantity` I replaced it in `Order` and used it in `get_trades` `Trade` constructor in place of calculation `item['Quantity'] - item['QuantityRemaining']))`
3. `PricePerUnit` field is depracated in `get_trades` API response I replaced it with `limit`

### CODE REVIEW

File: pyexchange/bittrex.py [hunk 1]
```diff
@@ -174,25 +93,21 @@ def place_order(self, pair: str, is_sell: bool, price: Wad, amount: Wad) -> str:
...
 
-        params = {
-            "quantity": str(amount),
-            "rate": str(price),
-            "market": pair
+        body = {
+            'marketSymbol': pair,
+            'direction': "SELL" if is_sell else "BUY",
+            'quantity': float(price),
+            'limit': float(price),
```
Comment (ith-harvey) [inline pyexchange/bittrex.py:100]: did you experiment turning this into a string rather than a float? float can cause some decimal issues.

**Review by ith-harvey — COMMENTED** [review]
Curious to hear a response about floats...

File: pyexchange/bittrex.py [hunk 2]
```diff
@@ -28,110 +29,28 @@
 from urllib.parse import urlencode
 
 from pyexchange.api import PyexAPI
+from pyexchange.model import Order as BaseOrder, Trade
```
Comment (MikeHathaway) [inline pyexchange/bittrex.py:32]: BaseOrder doesn't seem to be defined here?

File: pyexchange/bittrex.py [hunk 3]
```diff
@@ -148,23 +67,23 @@ def __init__(self, api_server: str, api_key: str, secret_key: str, timeout: floa
...
+    def get_precision(self, pair: str):
+        assert(isinstance(pair, str))
+        return self._http_request("GET", f"/v3/markets/{pair}", {})["precision"]
 
     def get_pair(self, pair: str):
         assert(isinstance(pair, str))
-        return next(filter(lambda symbol: symbol['MarketName'] == pair, self.get_markets()))
+        return next(filter(lambda market: market['symbol'] == pair, self.get_markets()))
 
     def get_balances(self):
```
Comment (MikeHathaway) [inline pyexchange/bittrex.py:80]: Minor point, but can you add the return Types for these functions? Helps with jumping into the codebase later

Comment (MikeHathaway) [reply pyexchange/bittrex.py]: Oh awesome find! Yeah that would be a great addition.

**Review by MikeHathaway — APPROVED** [review]
LGTM!

**Pull request merged**